### ISSUE TITLE
Parameter adjustments

### ISSUE THREAD

**Issue by gabrielguerrer** [InsaneMonster/NistRng#8]
Why do you hardcode the following parameters? It's more interesting to let the users pass them as arguments to the test functions.

      P a r a m e t e r   A d j u s t m e n t s
        -----------------------------------------
    [1] Block Frequency Test - block length(M):         128
    [2] NonOverlapping Template Test - block length(m): 9
    [3] Overlapping Template Test - block length(m):    9
    [4] Approximate Entropy Test - block length(m):     10
    [5] Serial Test - block length(m):                  16
    [6] Linear Complexity Test - block length(M):       500

**Comment by yagoiroot** [comment]
@gabrielguerrer I agree that having the ability to control the parameters would be much more useful. 

In addition to that, it appears that the values for the hardcoded parameters are not necessarily correct for some of the tests.
